## Air Conditioners Data Analysis

**Stage 8: Results & Interpretation**

This notebook is independently executable in Google Colab. Every step explains what we do, why we do it, what we obtain, and how the result is used in the next stage.


## Stage Objective

Stage 8 brings together the dataset profile, analysis results and visual evidence into final project findings.


In [11]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Load the project CSV directly so this notebook is independently executable.
csv_path = "/content/Air_Conditioners_Cleaned.csv"
df = pd.read_csv(csv_path)

print("CSV loaded successfully.")
print("Dataset shape:", df.shape)


CSV loaded successfully.
Dataset shape: (720, 14)


## Why This Stage Matters

A project is not complete when charts are generated. The final stage explains what the charts and statistics mean and how they connect to the original objective.


In [12]:
print('Missing values:',int(df.isna().sum().sum()))

Missing values: 217


## Reload the CSV

The final notebook loads the same supplied CSV directly so the reported dataset size and quality can be verified.


In [13]:
print('Duplicate rows:',int(df.duplicated().sum()))

Duplicate rows: 0


## Final Dataset Profile

The project is based on **720 rows and 14 columns** in the supplied CSV. This profile anchors the conclusions to the actual dataset.


In [14]:
nums=df.select_dtypes(include=np.number).columns.tolist(); print(df[nums].describe().T)

                  count          mean           std           min  \
ratings           720.0      3.888194      0.612151      1.000000   
no_of_ratings     720.0    151.262500    562.513005      1.000000   
discount_price    720.0  40183.979167  11938.526137    199.000000   
actual_price      720.0  55479.741139  18984.776512    499.000000   
discount_amount   720.0  15295.761972  13102.876643 -38541.000000   
discount_percent  720.0     -2.351720    262.575704  -5938.520801   
tonnage           626.0      1.468164      0.501664      0.750000   
energy_star       597.0      3.577889      0.951534      1.000000   

                           25%          50%           75%           max  
ratings               3.900000      4.00000      4.000000       5.00000  
no_of_ratings         8.000000     18.00000     40.000000    9577.00000  
discount_price    37282.500000  39190.00000  41999.250000  128800.00000  
actual_price      51472.500000  56990.00000  61990.000000  149000.00000  
discount

## Final Quality Review

Missing-value and duplicate checks are repeated so the final discussion is based on a verified data state.


In [15]:
cats=df.select_dtypes(include=['object','category','bool']).columns.tolist()
for c in cats[:5]: print(c,df[c].value_counts().head(10).to_dict())

name {'Samsung 1.5 Ton 3 Star Wind-Free Technology Inverter Split AC (Copper, Convertible 5-in-1 Cooling Mode, Anti-bacterial Fil...': 3, 'White Westinghouse 1.5 ton 3 Star fix Speed Split AC': 3, 'LG 1.5 Ton 5 Star AI DUAL Inverter Split AC (Copper, Super Convertible 6-in-1 Cooling, HD Filter with Anti-Virus Protectio...': 2, 'VESTAR 1.0Ton 3 Star Split AC (100% Copper Coil with Hidden Display Eco/Power Saving Mode, White)': 2, 'Samsung 1.5 Ton 3 Star Wi-fi Enabled, Wind-Free Technology Inverter Split AC (Copper, Convertible 5-in-1 Cooling Mode, Ant...': 2, 'LG 2.0 Ton 3 Star AI DUAL Inverter Split AC (Copper, Super Convertible 6-in-1 Cooling, 4 Way Swing, HD Filter with Anti-Vi...': 2, 'Panasonic 1 Ton 5 Star Wi-Fi Inverter Smart Split AC (Copper, 7 in 1 Convertible with additional AI Mode, 4 Way Swing, nan...': 2, 'Personal Air Conditioners Portable AC Mini Air Conditioner for Office Dorm Room,Small AC Desktop Air Cooler Fan with Humid...': 2, 'Panasonic 1.5 Ton 5 Star Wi-Fi Inverte

## Key Numerical Findings

The main descriptive statistics and group summaries are reviewed to identify the strongest observable patterns.


In [16]:
print(df['discount_price'].describe().to_dict() if 'discount_price' in df.columns else 'Price field not detected')

{'count': 720.0, 'mean': 40183.979166666664, 'std': 11938.526137340927, 'min': 199.0, '25%': 37282.5, '50%': 39190.0, '75%': 41999.25, 'max': 128800.0}


## Key Category Findings

Category counts and group-level measures are interpreted together rather than looking at one statistic in isolation.


In [17]:
print(df['ratings'].describe().to_dict() if 'ratings' in df.columns else 'Rating field not detected')

{'count': 720.0, 'mean': 3.8881944444444443, 'std': 0.6121505591456029, 'min': 1.0, '25%': 3.9, '50%': 4.0, '75%': 4.0, 'max': 5.0}


## Visual Evidence

The charts from Stage 7 are connected to the corresponding numerical results, showing how each interpretation was obtained.


In [18]:
print('Final interpretation connects numerical summaries with the visual evidence from Stage 7.')

Final interpretation connects numerical summaries with the visual evidence from Stage 7.


## Practical Meaning

The analysis provides a structured understanding of the air-conditioner products represented in the dataset, including observable differences in available specifications and commercial attributes.


In [19]:
print('Limitations: conclusions depend on the supplied dataset and available attributes.')

Limitations: conclusions depend on the supplied dataset and available attributes.


## Limitations and Future Scope

The conclusions depend on the supplied dataset, its available attributes and its represented products. Future work could add more data, interactive dashboards, richer features and predictive analysis.


In [20]:
print('Future scope: larger data, dashboards, richer features and predictive modelling.')

Future scope: larger data, dashboards, richer features and predictive modelling.
